.github/workflows/scripts/build_npm.ts

```diff
@@ -0,0 +1,81 @@
+import { build, emptyDir } from "https://deno.land/x/dnt@0.22.0/mod.ts";
+import { parse } from "https://deno.land/std@0.133.0/flags/mod.ts";
+
+const entryPointDefault = "./mod.ts";
+const outDirDefault = "./.github/workflows/dist";
+
+const helpText =
+  "Deno to node transpiler, converts the ts_serialize module to a node compatible module, and writes it to .github/workflows/npm, or the directory of your choice." +
+  "\n" +
+  "\nThis must be run from the project root if using the default arguments." +
+  "\n" +
+  "\nUsage:" +
+  `\n\tdeno run -A .github/workflows/scripts/build_npm.ts -v "v0.0.0-test" [-e ${entryPointDefault}] [-o ${outDirDefault}]` +
+  "\n\tdeno run -A .github/workflows/scripts/build_npm.ts --help" +
+  "\n" +
+  "\nCommand line arguments:" +
+  "\n\t-h,  --help               Prints this help message, then exits." +
+  "\n\t-v,  --version=[version]  The version value to set in the resulting package.json. Required." +
+  `\n\t-e,  --entry-point=[path] The path to the mod.ts file to transpile into the npm package. Defaults to "${entryPointDefault}"` +
+  `\n\t-o,  --out=[path]         The path to the target directory where the built files should be placed. Defaults to "${outDirDefault}"`;
+
+function printHelpText() {
+  console.log(helpText);
+}
+
+async function main() {
+  const flags = parse(Deno.args);
+
+  if (flags["h"] || flags["help"]) {
+    printHelpText();
+    return;
+  }
+
+  const version = flags["v"] || flags["version"];
+  const entryPoint = flags["e"] || flags["entry-point"] || entryPointDefault;
+  const outDir = flags["o"] || flags["out"] || outDirDefault;
+
+  if (!version) {
+    printHelpText();
+    return 1;
+  }
+
+  await emptyDir(outDir);
+
+  await build({
+    entryPoints: [entryPoint],
+    outDir,
+    compilerOptions: { target: "ES2021" },
+    shims: {
+      // see JS docs for overview and more options
+      deno: true,
+    },
+    package: {
+      name: "@gamebridgeai/ts_serialize",
+      version,
+      description: "A zero dependency library for serializing data.",
+      repository: {
+        type: "git",
+        url: "https://github.com/GameBridgeAI/ts_serialize.git",
+      },
+      author: "GameBridgeAI",
+      license: "MIT",
+      bugs: {
+        url: "https://github.com/GameBridgeAI/ts_serialize/issues",
+      },
+      homepage: "https://gamebridgeai.github.io/ts_serialize",
+      keywords: [
+        "typescript",
+        "serialize",
+        "serialization",
+        "JSON",
+        "node",
+        "deno",
```

Comment:
let's remove these, pretty uncommon search terms

Reply:
I wouldn't say they are uncommon, personally I am fine with keeping them. If someone searches for "node serialization" then this would be picked up.

Looking at other serialization projects I think we can actually add in `deserialize`. I think removing `serialization` is probably fine though, as it overlaps with `serialize`

Reply:
cool, let's do that then!